pkg/controller/operands/networkAddons_test.go

```diff
@@ -268,6 +270,195 @@ var _ = Describe("CNA Operand", func() {
 			Expect(req.Conditions).To(BeEmpty())
 		})
 
+		It("should add self signed configutation if missing in CNAO", func() {
+			existingResource, err := NewNetworkAddons(hco)
+			Expect(err).ToNot(HaveOccurred())
+
+			hco.Spec.CertConfig = &hcov1beta1.HyperConvergedCertConfig{
+				CA: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 24 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 1 * time.Hour},
+				},
+				Server: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 12 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 30 * time.Minute},
+				},
+			}
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Updated).To(BeTrue())
+			Expect(res.Err).To(BeNil())
+
+			foundResource := &networkaddonsv1.NetworkAddonsConfig{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			Expect(foundResource.Spec.SelfSignConfiguration).ToNot(BeNil())
+			selfSignedConfig := foundResource.Spec.SelfSignConfiguration
+			Expect(selfSignedConfig.CARotateInterval).Should(Equal("24h0m0s"))
+			Expect(selfSignedConfig.CAOverlapInterval).Should(Equal("1h0m0s"))
+			Expect(selfSignedConfig.CertRotateInterval).Should(Equal("12h0m0s"))
+			Expect(selfSignedConfig.CertOverlapInterval).Should(Equal("30m0s"))
+
+			Expect(req.Conditions).To(BeEmpty())
+		})
+
+		It("should set self signed configutation to defaults if missing in HCO CR", func() {
+			dummyHCO := commonTestUtils.NewHco()
+
+			dummyHCO.Spec.CertConfig = &hcov1beta1.HyperConvergedCertConfig{
+				CA: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 24 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 1 * time.Hour},
+				},
+				Server: &hcov1beta1.CertRotateConfig{
+					Duration:    metav1.Duration{Duration: 12 * time.Hour},
+					RenewBefore: metav1.Duration{Duration: 30 * time.Minute},
+				},
+			}
```

Comment:
I fear that this doesn't really match the test description

Reply:
Done